### PR TITLE
NativeAOT-LLVM: Add System.Globalization.Native.

### PR DESCRIPTION
**Pull request by yowl**
This PR adds the libSystem.Globalization.Native.a library to the package and link step.
Also adds the ILLink substitution settings to remove a few undefined symbols.  This gets the list of undefined symbols for samples\HelloWorld to:
```
EXEC : warning : undefined symbol: dotnet_browser_entropy (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
```
For some reason `libSystem.Globalization.Native.a` was going to a lib subfolder, so there's a `CMakeLists.txt` fix for that - not sure why it is done differently in the first place, but this change is necessary to get the archive picked up for packaging.

### CODE REVIEW

File: src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj [hunk 1]
```diff
@@ -5,6 +5,9 @@
   <ItemGroup Condition="'$(TargetsUnix)'=='true'">
     <ILLinkSubstitutionsXmls Include="ILLink\ILLink.Substitutions.Unix.xml" />
   </ItemGroup>
+  <ItemGroup Condition="'$(TargetsBrowser)'=='true'">
+    <ILLinkSubstitutionsXmls Include="ILLink\ILLink.Substitutions.llvm.xml" />
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/System.Private.CoreLib.csproj:9]: This should be `.Browser.xml`

File: src/coreclr/nativeaot/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.llvm.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      <method signature="System.Int32 get_ProcessorCount()" body="stub" value="1" />
+    </type>
+    <type fullname="System.Runtime.CompilerServices.RuntimeFeature">
+      <method signature="System.Boolean get_IsDynamicCodeCompiled()" body="stub" value="false" />
+    </type>
+    <type fullname="System.Resources.ResourceReader">
+      <method signature="System.Boolean InitializeBinaryFormatter()" body="stub" value="false" />
+    </type>
+    <type fullname="System.Globalization.GlobalizationMode">
+      <method signature="System.Boolean get_UseNls()" body="stub" value="false" />
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.llvm.xml:16]: Is this going unconditionally trim the full globalization support? Do we even need `System.Globalization.Native` in that case?

Comment (yowl) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.llvm.xml:16]: Well if I remove the library from the linking, then I get back
```
EXEC : warning : undefined symbol: GlobalizationNative_ChangeCase (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_ChangeCaseInvariant (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_ChangeCaseTurkish (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_CloseSortHandle (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_CompareString (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_EndsWith (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetCalendarInfo (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetCalendars (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetDefaultLocaleName (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetJapaneseEraStartDate (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLatestJapaneseEra (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleInfoGroupingSizes (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleInfoInt (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleInfoString (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleName (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetLocaleTimeFormat (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_GetSortHandle (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_IndexOf (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_InitOrdinalCasingPage (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_LastIndexOf (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
EXEC : warning : undefined symbol: GlobalizationNative_StartsWith (referenced by top-level compiled C/C++ code) [E:\GitHub\runtimelab\samples\HelloWorld\HelloWorld.csproj]
```
So it seems there are still some things left in.  `GlobalizationNative_StartsWith` for example is rooted up through a few calls to `S_P_CoreLib_System_Globalization_CompareInfo__IcuStartsWith` and then `string.StartsWith`  I could go through the code seeing if there's some other substitution that should be added?

Comment (jkotas) [reply src/coreclr/nativeaot/System.Private.CoreLib/src/ILLink/ILLink.Substitutions.llvm.xml]: Ah, I have confused Nls and Icu. This makes sense.

File: src/libraries/Native/Unix/System.Globalization.Native/CMakeLists.txt [hunk 3]
```diff
@@ -103,7 +104,11 @@ else()
     # it needs to build with coreclr in order to be linkable with it on Windows
     # coreclr and static libraries must match because linker does not permit mixing modules
     # with different C runtimes (i.e. Debug vs. Release)
-    install (TARGETS System.Globalization.Native-Static DESTINATION lib)
+    if(CLR_CMAKE_TARGET_UNIX_WASM)
```
Comment (jkotas) [inline src/libraries/Native/Unix/System.Globalization.Native/CMakeLists.txt:107]: Other libraries shims use `DESTINATION ${STATIC_LIB_DESTINATION}` for the static libraries. Would it fix the problem?

Comment (jkotas) [reply src/libraries/Native/Unix/System.Globalization.Native/CMakeLists.txt:107]: I am trying to clean this up in https://github.com/dotnet/runtime/pull/47127 . We can merge this change in the meantime.

Comment (yowl) [reply src/libraries/Native/Unix/System.Globalization.Native/CMakeLists.txt:107]: Thanks!

**Pull request merged**